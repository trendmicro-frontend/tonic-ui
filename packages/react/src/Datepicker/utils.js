import {
  FEB_OF_LEAP_YEAR,
  MONTH_DAYS,
} from './Calendar/constants';
import { SEPARATOR } from './constants';

export const convertToDateObj = (data) => {
  if (typeof data === 'string') {
    // We didn't use date strings to create a Date object due to browser differences and inconsistencies
    return new Date(...data.split(SEPARATOR).map((val, index) => (index === 1 ? +val - 1 : +val)));
  }
  if (typeof data === 'number') {
    return new Date(data);
  }
  return data;
};

export const dateToAry = (rawDate) => {
  let newDate = convertToDateObj(rawDate);
  if (!(newDate instanceof Date)) {
    return [];
  }
  const [month, date, year] = newDate.toLocaleDateString('en-US').split('/');
  return [+year, +month, +date];
};

export const dateToStrAry = (rawDate) => {
  return dateToAry(rawDate).map((item) => (item < 10 ? `0${item}` : `${item}`));
};

export const dateToObj = (rawDate) => {
  const [year, month, date] = dateToAry(rawDate);
  return { year, month, date };
};

export const dateToStr = (rawDate) => {
  return dateToStrAry(rawDate).join('-');
};

export const isValidDate = (dateStr) => !!new Date(dateStr).toJSON();

export const getTimestamp = (date) => {
  const dateObj = convertToDateObj(date);
  return dateObj.getTime();
};

export const isLeapYear = (checkYear) => new Date(checkYear, 1, FEB_OF_LEAP_YEAR).getDate() === FEB_OF_LEAP_YEAR;

export const getMonthDays = (year, month) => (isLeapYear(year) && month === 2
  ? FEB_OF_LEAP_YEAR
  : MONTH_DAYS[month - 1]);

export const getPrevMonthLastDate = (curDate) => {
  curDate.setDate(0);
  return curDate;
};

/**
 * Uses canvas.measureText to compute and return the width of the given text of given font in pixels.
 *
 * @param text The text to be rendered.
 * @param {String} font The css font descriptor that text is to be rendered with (e.g. '14px verdana').
 *
 * @see http://stackoverflow.com/questions/118241/calculate-text-width-with-javascript/21015393#21015393
 */
export const getTextWidth = (text, font) => {
  if (typeof document === 'undefined') {
    return 0;
  }
  // if given, use cached canvas for better performance
  // else, create new canvas
  let canvas = getTextWidth.canvas ||
    (getTextWidth.canvas = document.createElement('canvas')) || null;
  let context = canvas.getContext('2d');
  context.font = font;
  let metrics = context.measureText(text);
  return metrics.width;
};
