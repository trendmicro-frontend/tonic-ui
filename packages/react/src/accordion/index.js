import Accordion from './Accordion';
import AccordionBody from './AccordionBody';
import AccordionCollapse from './AccordionCollapse';
import AccordionHeader from './AccordionHeader';
import AccordionItem from './AccordionItem';
import AccordionToggle from './AccordionToggle';
import AccordionToggleIcon from './AccordionToggleIcon';
import useAccordion from './useAccordion';
import useAccordionItem from './useAccordionItem';

Accordion.Body = AccordionBody;
Accordion.Collapse = AccordionCollapse;
Accordion.Header = AccordionHeader;
Accordion.Item = AccordionItem;
Accordion.Toggle = AccordionToggle;
Accordion.Toggle.Icon = AccordionToggleIcon;

export {
  Accordion,
  AccordionBody,
  AccordionCollapse,
  AccordionHeader,
  AccordionItem,
  AccordionToggle,
  AccordionToggleIcon,
  useAccordion,
  useAccordionItem,
};
