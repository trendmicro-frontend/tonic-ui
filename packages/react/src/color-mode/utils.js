const matchMediaQuery = (query) => {
  const mediaQueryList = window?.matchMedia?.(query);
  if (!mediaQueryList) {
    return undefined;
  }
  return mediaQueryList.matches;
};

export const colorSchemeQuery = {
  dark: '(prefers-color-scheme: dark)',
  light: '(prefers-color-scheme: light)',
};

export const getColorScheme = (fallbackColorMode) => {
  const isDarkColorScheme = matchMediaQuery(colorSchemeQuery.dark) ?? (fallbackColorMode === 'dark');
  return isDarkColorScheme ? 'dark' : 'light';
};
