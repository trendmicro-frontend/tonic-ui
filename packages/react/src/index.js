// components
export * from './accordion';
export * from './alert';
export * from './badge';
export * from './box';
export * from './button';
export * from './checkbox';
export * from './color-mode';
export * from './color-style';
export * from './css-baseline';
export * from './divider';
export * from './drawer';
export * from './flat-button';
export * from './flex';
export * from './grid';
export * from './icon';
export * from './image';
export * from './input';
export * from './link';
export * from './menu';
export * from './modal';
export * from './pagination';
export * from './popover';
export * from './popper';
export * from './portal';
export * from './presence';
export * from './radio';
export * from './search-input';
export * from './scrollbar';
export * from './select';
export * from './skeleton';
export * from './space';
export * from './spinner';
export * from './stack';
export * from './switch';
export * from './table';
export * from './tabs';
export * from './tag';
export * from './text';
export * from './textarea';
export * from './theme';
export * from './toast';
export * from './tooltip';
export * from './transitions';
export * from './visually-hidden';

// hooks
export useDisclosure from './deprecated/useDisclosure'; // deprecated

// settings
export colorStyle from './shared/color-style';
export theme from './shared/theme';
