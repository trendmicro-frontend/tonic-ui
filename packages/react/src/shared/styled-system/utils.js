import { get } from '@styled-system/core';

/**
 * The `width` prop is transformed based on the following:
 * • Numbers from 0-1 are converted to percentage widths.
 * • Numbers greater than 1 are converted to pixel values.
 * • String values are passed as raw CSS values.
 * • Arrays are converted to responsive width styles.
 * • If `theme.sizes` is defined, the `width` prop will attempt to pick up values from the theme.
 */
const getWidth = (n, scale) => {
  const defaultValue = (!Number.isFinite(n) || n > 1)
    ? n
    : n * 100 + '%';
  return get(scale, n, defaultValue);
};

export {
  getWidth,
};
