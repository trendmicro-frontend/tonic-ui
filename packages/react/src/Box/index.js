import styled from '@emotion/styled';
import isPropValid from '@emotion/is-prop-valid';
import { ensureArray } from 'ensure-type';
import { cx, system } from '../shared/styled-system';

const shouldForwardProp = (() => {
  const stylePropMap = ensureArray(system.propNames)
    .reduce((acc, val) => {
      acc[val] = true;
      return acc;
    }, {});
  const omittedStylePropMap = {
    ...stylePropMap,

    // The `as` prop is supported by Emotion
    'as': true,
  };

  return prop => isPropValid(prop) && !omittedStylePropMap[prop];
})();

const createPseudoClassTransformFunction = name => prop => {
  const result = {};
  if (Array.isArray(prop)) {
    result[`${name}(${prop[0]})`] = prop[1];
  } else if (typeof prop === 'object') {
    for (const key in prop) {
      if (Object.prototype.hasOwnProperty.call(prop, key)) {
        result[`${name}(${key})`] = prop[key];
      }
    }
  }
  return result;
};

/**
 * Pseudo-classes
 * https://developer.mozilla.org/en-US/docs/Web/CSS/
 */

const active = [
  '&:active',
  '&[data-active=true]'
].join(',');
const checked = [
  '&[aria-checked=true]',
  '&:checked',
].join(',');
const selected = [
  '&[aria-selected=true]',
  '&[data-selected=true]',
  '&:active[data-selected=true]',
  '&:hover[data-selected=true]',
].join(',');
const disabled = [
  '&[aria-disabled=true]',
  '&:disabled',
  '&:disabled:hover',
  '&:disabled:focus',
  '&:hover[aria-disabled=true]',
  '&:focus[aria-disabled=true]',
].join(',');
const empty = '&:empty';
const enabled = [
  '&:enabled',
  '&:enabled:focus',
  '&:enabled:hover',
].join(',');
const firstChild = '&:first-child';
const firstOfType = '&:first-of-type';
const fullscreen = '&:fullscreen';
const focus = '&:focus';
const focusActive = '&:focus:active';
const focusHover = '&:focus:hover';
const focusWithin = '&:focus-within';
const focusSelected = ['&[aria-selected=true]:focus', '&[data-selected=true]:focus'].join(',');
const hover = '&:hover';
const indeterminate = '&:indeterminate';
const valid = [
  '&[aria-invalid=false]',
  '&:valid',
].join(',');
const invalid = [
  '&[aria-invalid=true]',
  '&:invalid',
].join(',');
const lastChild = '&:last-child';
const lastOfType = '&:last-of-type';
const nthOfTypeFn = createPseudoClassTransformFunction('&:nth-of-type');
const readOnly = [
  '&[aria-readonly=true]',
  '&:read-only',
].join(',');
const visited = '&:visited';

/**
 * Pseudo-elements
 * https://developer.mozilla.org/en-US/docs/Web/CSS/
 */

const after = '&::after';
const backdrop = '&::backdrop';
const before = '&::before';
const cue = '&::cue';
const firstLetter = '&::first-letter';
const firstLine = '&::first-line';
const placeholder = '&::placeholder';
const selection = '&::selection';

const _Box = styled('div', {
  shouldForwardProp,
})(system);

const Box = styled(_Box)(
  ({
    // pseudo-classes
    _active,
    _checked,
    _selected,
    _disabled,
    _empty,
    _enabled,
    _firstChild,
    _firstOfType,
    _fullscreen,
    _focus,
    _focusActive,
    _focusHover,
    _focusWithin,
    _focusSelected,
    _hover,
    _indeterminate,
    _valid,
    _invalid,
    _lastChild,
    _lastOfType,
    _nthOfType,
    _readOnly,
    _visited,

    // pseudo-elements
    __after,
    __backdrop,
    __before,
    __cue,
    __firstLetter,
    __firstLine,
    __placeholder,
    __selection,
  }) => {
    let rest = null;

    if (_nthOfType) {
      rest = { ...rest, ...nthOfTypeFn(_nthOfType) };
    }

    return cx({
      /**
       * Pseudo-classes must be declared in a specific order, as shown below:
       *
       * ```
       * :link
       * :visited
       * :hover
       * :active
       * ```
       *
       * Each pseudo-class corresponds to an event which can only happen later in the timeline than the one before.
       *
       * That is to say:
       *
       * 1. A link is unvisited before it is visited.
       * 2. A link is visited before it is hovered over.
       * 3. A link is hovered over before it is in active use.
       */
      [focus]: _focus,
      [visited]: _visited,
      [hover]: _hover,
      [focusHover]: _focusHover,
      [active]: _active,
      [focusActive]: _focusActive,
      [focusSelected]: _focusSelected,
      [focusWithin]: _focusWithin,
      [checked]: _checked,
      [selected]: _selected,
      [disabled]: _disabled,
      [empty]: _empty,
      [enabled]: _enabled,
      [firstChild]: _firstChild,
      [firstOfType]: _firstOfType,
      [fullscreen]: _fullscreen,
      [indeterminate]: _indeterminate,
      [valid]: _valid,
      [invalid]: _invalid,
      [lastChild]: _lastChild,
      [lastOfType]: _lastOfType,
      [readOnly]: _readOnly,

      // pseudo-elements
      [after]: __after,
      [backdrop]: __backdrop,
      [before]: __before,
      [cue]: __cue,
      [firstLetter]: __firstLetter,
      [firstLine]: __firstLine,
      [placeholder]: __placeholder,
      [selection]: __selection,

      ...rest
    });
  },
);

Box.displayName = 'Box';

export default Box;
