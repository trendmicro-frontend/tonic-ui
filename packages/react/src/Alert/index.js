import React, { forwardRef } from 'react';
import Box from '../Box';
import ButtonBase from '../ButtonBase';
import Flex from '../Flex';
import Icon from '../Icon';
import Space from '../Space';
import {
  useAlertStyle,
  useAlertIconStyle,
  useAlertMessageStyle,
  useAlertCloseButtonStyle,
} from './styles';

const defaultVariant = 'solid';
const defaultSeverity = 'success';

const getIconBySeverity = (severity) => {
  const iconName = {
    success: 'success',
    info: 'info',
    warning: 'warning-triangle',
    error: 'error',
  }[severity];

  if (!iconName) {
    return null;
  }

  return (
    <Icon icon={`${iconName}`} />
  );
};

const AlertIcon = (props) => (
  <Flex {...props} />
);

const AlertMessage = (props) => (
  <Box {...props} />
);

const AlertCloseButton = (props) => (
  <ButtonBase {...props} />
);

const Alert = forwardRef((
  {
    isClosable: _isClosable = false,
    isCloseButtonVisible: LEGACY_isCloseButtonVisible = false, // eslint-disable-line camelcase
    onClose,
    variant = defaultVariant,
    severity = defaultSeverity,
    icon,
    children,
    ...rest
  },
  ref,
) => {
  const isClosable = _isClosable || LEGACY_isCloseButtonVisible; // eslint-disable-line camelcase
  const styleProps = useAlertStyle({ variant, severity });
  const iconStyleProps = useAlertIconStyle({ variant, severity });
  const messageStyleProps = useAlertMessageStyle();
  const closeButtonStyleProps = useAlertCloseButtonStyle({ variant });

  if (typeof icon === 'string') {
    icon = (<Icon icon={icon} />);
  }
  if (typeof icon === 'undefined') {
    icon = getIconBySeverity(severity);
  }

  return (
    <Flex
      ref={ref}
      align="flex-start"
      justify="space-between"
      {...styleProps}
      {...rest}
    >
      {!!icon && (
        <>
          <AlertIcon {...iconStyleProps}>
            {icon}
          </AlertIcon>
          <Space minWidth="2x" />
        </>
      )}
      <AlertMessage {...messageStyleProps}>
        {children}
      </AlertMessage>
      {!!isClosable && (
        <>
          <Space minWidth="4x" />
          <AlertCloseButton {...closeButtonStyleProps} onClick={onClose}>
            <Icon icon="close-s" />
          </AlertCloseButton>
        </>
      )}
    </Flex>
  );
});

Alert.displayName = 'Alert';

export default Alert;
