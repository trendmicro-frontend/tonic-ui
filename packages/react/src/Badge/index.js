import React, { forwardRef } from 'react';
import useBadgeStyle from './styles';
import Box from '../Box';

const Badge = forwardRef(
  (
    {
      variant = 'badge',
      variantColor = 'red',
      badgeContent,
      isHidden = false,
      children,
      offset,
      dotSize = 8,
      borderColor,
      borderWidth = 1,
      ...restProps
    },
    ref,
  ) => {
    const badgeBorderColor = borderColor ?? variantColor;
    const badgeStyleProps = useBadgeStyle({
      color: variantColor,
      hasChildren: !!children,
      showAsDot: variant === 'dot',
      offset,
      dotSize,
      borderColor: badgeBorderColor,
      borderWidth
    });

    return (
      <Box
        as="span"
        position="relative"
        display="inline-block"
      >
        {children}
        {!isHidden && (
          <Box
            ref={ref}
            as="span"
            fontWeight="normal"
            borderRadius="sm"
            whiteSpace="nowrap"
            {...badgeStyleProps}
            {...restProps}
          >
            {badgeContent}
          </Box>
        )}
      </Box>
    );
  }
);

Badge.displayName = 'Badge';

export default Badge;
