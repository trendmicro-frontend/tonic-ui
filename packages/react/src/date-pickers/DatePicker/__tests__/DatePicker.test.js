import { screen, waitForElementToBeRemoved } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { render } from '@tonic-ui/react/test-utils/render';
import { transitionDuration } from '@tonic-ui/utils/src';
import { testA11y } from '@tonic-ui/react/test-utils/accessibility';
import {
  Box,
  DatePicker,
  InputAdornment,
  InputControl,
  Text,
  useColorStyle,
} from '@tonic-ui/react/src';
import { CalendarIcon } from '@tonic-ui/react-icons/src';
import * as dateFns from 'date-fns';
import React, { useCallback } from 'react';

describe('DatePicker', () => {
  const TestComponent = (props) => {
    const [colorStyle] = useColorStyle();
    const inputFormat = 'MM/dd/yyyy';
    const onChange = jest.fn();
    const onError = jest.fn();
    const inputError = false;
    const value = new Date('2024-08-01');
    const formatDate = useCallback((date, format) => {
      return dateFns.format(date, format);
    }, []);
    const renderInput = useCallback(({ error, inputProps }) => {
      return (
        <Box>
          <InputControl
            {...inputProps}
            startAdornment={(
              <InputAdornment color={colorStyle.color.secondary}>
                <CalendarIcon />
              </InputAdornment>
            )}
            data-testid="date-picker-input"
            error={inputError}
            placeholder={inputFormat}
          />
          {inputError && (
            <Text mt="1x" color="red:50">Invalid date</Text>
          )}
        </Box>
      );
    }, [colorStyle, inputError, inputFormat]);

    return (
      <DatePicker
        data-testid="date-picker"
        closeOnSelect={false}
        firstDayOfWeek={0}
        formatDate={formatDate}
        onChange={onChange}
        onError={onError}
        value={value}
        inputFormat={inputFormat}
        renderInput={renderInput}
        {...props}
      />
    );
  };

  it('should render correctly', async () => {
    const user = userEvent.setup();
    const renderOptions = {
      useCSSVariables: true,
    };
    const { container } = render((
      <TestComponent />
    ), renderOptions);

    const datePicker = screen.getByTestId('date-picker');
    const datePickerInput = screen.getByTestId('date-picker-input');

    // The date picker and date picker input should be in the document
    expect(datePicker).toBeInTheDocument();
    expect(datePickerInput).toBeInTheDocument();

    // Open the date picker
    await user.click(datePickerInput);

    // The "menu" role should be in the document
    expect(await screen.findByRole('menu')).toBeInTheDocument();

    expect(container).toMatchSnapshot();

    await testA11y(container, {
      axeOptions: {
        rules: {
          // FIXME: Certain ARIA roles must contain particular children (aria-required-children)"
          'aria-required-children': { enabled: false },

          // FIXME: Interactive controls must not be nested (nested-interactive)
          'nested-interactive': { enabled: false },
        },
      },
    });
  });

  it('should close the date picker when a date is selected and closeOnSelect is true', async () => {
    const user = userEvent.setup();
    render(<TestComponent closeOnSelect={true} />);

    const datePickerInput = screen.getByTestId('date-picker-input');

    // Open the date picker
    await user.click(datePickerInput);

    // Select a date
    const dateButton = screen.getByText('15'); // Assuming 15th is a selectable date
    await user.click(dateButton);

    const duration = 100; // Shorten the duration to 100ms for testing
    // The "menu" role should not be in the document
    await waitForElementToBeRemoved(() => screen.getByRole('menu'), {
      // The toast should be removed after the duration plus the transition.
      timeout: duration + transitionDuration.standard + 100, // see "date-pickers/DatePicker/DatePickerContent.js"
    });
  });
});
