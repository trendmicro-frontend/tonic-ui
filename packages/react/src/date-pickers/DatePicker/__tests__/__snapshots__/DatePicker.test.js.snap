// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render correctly 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-size: var(--tonic-fontSizes-sm);
  line-height: var(--tonic-lineHeights-sm);
  height: var(--tonic-sizes-8x);
  background-color: var(--tonic-colors-transparent);
  border: var(--tonic-borders-1);
  border-color: var(--tonic-colors-gray-60);
  border-radius: var(--tonic-radii-sm);
  color: var(--tonic-colors-white-primary);
}

.emotion-4:invalid,
.emotion-4[aria-invalid=true] {
  border-color: var(--tonic-colors-red-50);
}

.emotion-4::-webkit-input-placeholder {
  color: var(--tonic-colors-white-tertiary);
  opacity: 1;
}

.emotion-4::-moz-placeholder {
  color: var(--tonic-colors-white-tertiary);
  opacity: 1;
}

.emotion-4:-ms-input-placeholder {
  color: var(--tonic-colors-white-tertiary);
  opacity: 1;
}

.emotion-4::placeholder {
  color: var(--tonic-colors-white-tertiary);
  opacity: 1;
}

.emotion-4:focus,
.emotion-4[data-focus] {
  border-color: var(--tonic-colors-blue-60);
  z-index: 1;
}

.emotion-4:hover,
.emotion-4[data-hover] {
  border-color: var(--tonic-colors-blue-50);
  z-index: 2;
}

.emotion-4:disabled,
.emotion-4:disabled:focus,
.emotion-4:disabled:hover,
.emotion-4[aria-disabled=true],
.emotion-4[aria-disabled=true]:focus,
.emotion-4[aria-disabled=true]:hover,
.emotion-4[data-disabled] {
  border-color: var(--tonic-colors-gray-60);
  cursor: not-allowed;
  opacity: .28;
}

.emotion-4:has(> input:focus:invalid),
.emotion-4:has(> input:focus[aria-invalid=true]) {
  border-color: var(--tonic-colors-blue-60);
}

.emotion-4:hover:has(> input:invalid),
.emotion-4:hover:has(> input[aria-invalid=true]) {
  border-color: var(--tonic-colors-red-50);
}

.emotion-4:hover:has(> input:focus:invalid),
.emotion-4:hover:has(> input:focus[aria-invalid=true]) {
  border-color: var(--tonic-colors-blue-60);
}

.emotion-6 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  padding-left: var(--tonic-space-3x);
  padding-right: var(--tonic-space-3x);
  color: var(--tonic-colors-white-secondary);
}

.emotion-8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: var(--tonic-sizes-4x);
  height: var(--tonic-sizes-4x);
  vertical-align: middle;
}

.emotion-10 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: inherit;
  border: none;
  color: inherit;
  outline: 0;
  padding: 0;
  width: 100%;
  font-size: var(--tonic-fontSizes-sm);
  line-height: var(--tonic-lineHeights-sm);
  padding-right: calc(.75rem - .0625rem);
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="date-picker"
  >
    <div
      class="emotion-2 emotion-1"
    >
      <div
        aria-controls="@tonic-ui/react:DatePickerContent-1"
        aria-expanded="false"
        aria-haspopup="menu"
        class="emotion-4 emotion-1"
        data-testid="date-picker-input-control"
        role="button"
        tabindex="0"
      >
        <div
          class="emotion-6 emotion-1"
        >
          <svg
            aria-hidden="true"
            class="emotion-8 emotion-1"
            data-icon="CalendarIcon"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 16 16"
          >
            <path
              d="M15 1h-2v-1h-3v1h-4v-1h-3v1h-2c-0.552 0-1 0.448-1 1v0 13c0 0.552 0.448 1 1 1v0h14c0.552 0 1-0.448 1-1v0-13c0-0.552-0.448-1-1-1v0zM15 4v3h-4v-3h4zM11 1h1v2h-1v-2zM10 4v3h-4v-3h4zM10 8v3h-4v-3h4zM4 1h1v2h-1v-2zM5 4v3h-4v-3h4zM1 8h4v3h-4v-3zM1 15v-3h4v3h-4zM6 15v-3h4v3h-4zM15 15h-4v-3h4v3zM11 11v-3h4v3h-4z"
            />
          </svg>
        </div>
        <input
          class="emotion-10 emotion-1"
          data-testid="date-picker-input-element"
          id="@tonic-ui/react:DatePickerToggle-1"
          placeholder="yyyy-MM-dd"
          value="2024-08-15"
        />
      </div>
    </div>
  </div>
</div>
`;
